example/example_f08.f90

```diff
@@ -1,228 +1,261 @@
-
-  symprec = 1e-6_dp
-
-
-    ! Rutile
-
-  print '("Rutile:")'
-  num_atom = 6
-  lattice(1,:) = (/ 4.0_dp, 0.0_dp, 0.0_dp /)
-  lattice(2,:) = (/ 0.0_dp, 4.0_dp, 0.0_dp /)
-  lattice(3,:) = (/ 0.0_dp, 0.0_dp, 2.6_dp /)
-  positions(1:3, 1) = (/ 0.0_dp, 0.0_dp, 0.0_dp /)
-  positions(1:3, 2) = (/ 0.5_dp, 0.5_dp, 0.5_dp /)
-  positions(1:3, 3) = (/ 0.3_dp, 0.3_dp, 0.0_dp /)
-  positions(1:3, 4) = (/ 0.7_dp, 0.7_dp, 0.0_dp /)
-  positions(1:3, 5) = (/ 0.2_dp, 0.8_dp, 0.5_dp /)
-  positions(1:3, 6) = (/ 0.8_dp, 0.2_dp, 0.5_dp /)
-  atom_types(1:6) = (/ 1, 1, 2, 2, 2, 2 /)
-  shifted = [0,0,0]
-
-  call write_syminfo( max_num_sym, num_atom, &
-       lattice, symprec, atom_types, positions, &
-       mesh, shifted, is_time_reversal )
-
-  ! FCC
-  print '("")'
-  print '("FCC:")'
-  num_atom = 1
-  lattice(1,:) = (/ 0.0_dp, 2.0_dp, 2.0_dp /)
-  lattice(2,:) = (/ 2.0_dp, 0.0_dp, 2.0_dp /)
-  lattice(3,:) = (/ 2.0_dp, 2.0_dp, 0.0_dp /)
-  positions(1:3, 1) = (/ 0.0_dp, 0.0_dp, 0.0_dp /)
-  atom_types(1) = 1
-  shifted = [0,0,0]
-
-  call write_syminfo( max_num_sym, num_atom, &
-       lattice, symprec, atom_types, positions, &
-       mesh, shifted, is_time_reversal )
-
-
-
-
-
-  print '(/,"D0_{24}:")'
-  a = 2.89_dp
-
-  lattice(1,:) = [ sqrt(3.0_dp), -1.0_dp, 0.0_dp]
-  lattice(2,:) = [ sqrt(3.0_dp),  1.0_dp, 0.0_dp]
-  lattice(3,:) = [ 0.0_dp      ,  0.0_dp, sqrt(32/3.0_dp)]
-
-
-  num_atom = 16
-  positions(:, 1) = [ 0.0_dp, 0.0_dp, 0.0_dp]
-  positions(:, 2) = [ 3.0_dp, 0.0_dp, 0.0_dp]
-  positions(:, 3) = [ 0.0_dp, 3.0_dp, 0.0_dp]
-  positions(:, 4) = [ 3.0_dp, 3.0_dp, 0.0_dp]
-  positions(:, 5) = [ 2.0_dp, 2.0_dp, 1.0_dp]
-  positions(:, 6) = [ 5.0_dp, 2.0_dp, 1.0_dp]
-  positions(:, 7) = [ 2.0_dp, 5.0_dp, 1.0_dp]
-  positions(:, 8) = [ 5.0_dp, 5.0_dp, 1.0_dp]
-  positions(:, 9) = [ 0.0_dp, 0.0_dp, 2.0_dp]
-  positions(:,10) = [ 3.0_dp, 0.0_dp, 2.0_dp]
-  positions(:,11) = [ 0.0_dp, 3.0_dp, 2.0_dp]
-  positions(:,12) = [ 3.0_dp, 3.0_dp, 2.0_dp]
-  positions(:,13) = [ 1.0_dp, 1.0_dp, 3.0_dp]
-  positions(:,14) = [ 4.0_dp, 1.0_dp, 3.0_dp]
-  positions(:,15) = [ 1.0_dp, 4.0_dp, 3.0_dp]
-  positions(:,16) = [ 4.0_dp, 4.0_dp, 3.0_dp]
-
-  positions(1:2,:num_atom) = positions(1:2,:num_atom)/6
-  positions(3,:num_atom) = positions(3,:num_atom)/4
-
-
-  atom_types(:num_atom) = [ 1, 2, 2, 2, 1, 2, 2, 2, 1, 2, 2, 2, 2, 2, 2, 1]
-  mesh = [ 12, 12, 6]
-  shifted = [1,1,1]
-
-  call write_syminfo( max_num_sym, num_atom, &
-  &     lattice, symprec, atom_types, positions, &
-  &    mesh, shifted, is_time_reversal )
+    use, intrinsic :: iso_c_binding
+    implicit none
+
+    call run_examples()
+
+contains
+
+    subroutine run_examples()
+        print '("Rutile:")'
+        call example_spg_get_dataset_rutile()
+        print '("")'
+        print '("FCC:")'
+        call example_spg_get_dataset_FCC()
+        print '("D0_{24}:")'
+        print '("")'
+        call example_spg_get_dataset_D0_24()
+    end subroutine run_examples
```

Comment:
Same comment here. Do we need a full list of examples for fortran? We already have the main list in the C one

Reply:
I don't get many questions about how to use Fortran APIs. If we have a list of C API examples, we may not need many examples for Fortran APIs. A pitfall of Fortran APIs is the different memory orders of arrays. Other important difference is how to use `spg_get_dataset`. In C, it requires to free the memory space of the structure, but in Fortran, `allocatable` arrays are used in the structure, therefore no need to free it explicitly. If these points can be described, the Fortran examples achieve their minimum purpose, I think.

Reply:
Fully agree 👍